pkg/processor/batchprocessor.go

```diff
@@ -89,14 +89,20 @@ func (s *segmentsBatch) poll() {
 				telemetry.T.SegmentSent(int64(len(batch)))
 			}
 			elapsed := time.Since(start)
-			log.Infof("Successfully sent batch of %d segments (%1.3f seconds)", len(batch), elapsed.Seconds())
-			for _, unprocessedSegment := range r.UnprocessedTraceSegments {
-				telemetry.T.SegmentRejected(1)
-				log.Errorf("Unprocessed segment: %v", unprocessedSegment)
-				log.Debug("Batch that contains unprocessed segments")
-				for i := 0; i < len(batch); i++ {
-					log.Debug(*batch[i])
+
+			if len(r.UnprocessedTraceSegments) != 0 {
+				log.Infof("Sent batch of %d segments but had %d Unprocessed segments (%1.3f seconds)", len(batch),
+					len(r.UnprocessedTraceSegments), elapsed.Seconds())
+				for _, unprocessedSegment := range r.UnprocessedTraceSegments {
+					telemetry.T.SegmentRejected(1)
+					log.Errorf("Unprocessed segment: %v", unprocessedSegment)
+					log.Debug("Batch that contains unprocessed segments")
+					for i := 0; i < len(batch); i++ {
+						log.Debug(*batch[i])
+					}
```

Comment:
Isn't this still N^2 logging? E.g. in the case of an entire batch of 50 segments being unprocessed, I believe this would still print out 52 messages/segment = 2,600 messages total.

IMO we either don't need to print out the entire batch ever, or we should just do it once outside the `UnprocessedTraceSegments` loop.

Reply:
We just print the Unprocessed segments here, see the test result:
```
2020-08-11T23:00:00-07:00 [Info] Successfully sent batch of 2 segments (0.077 seconds)
2020-08-11T23:00:01-07:00 [Info] Successfully sent batch of 2 segments (0.076 seconds)
2020-08-11T23:00:02-07:00 [Info] Successfully sent batch of 2 segments (0.081 seconds)
2020-08-11T23:00:03-07:00 [Info] Successfully sent batch of 2 segments (0.066 seconds)
2020-08-11T23:00:11-07:00 [Info] Successfully sent batch of 1 segments (0.067 seconds)
2020-08-11T23:00:12-07:00 [Info] Sent batch of 2 segments but had 1 Unprocessed segments (0.088 seconds)
2020-08-11T23:00:12-07:00 [Error] Unprocessed segment: {
  ErrorCode: "InvalidId",
  Id: "0",
  Message: "Invalid segment. ErrorCode: InvalidId"
}
2020-08-11T23:00:13-07:00 [Info] Sent batch of 2 segments but had 1 Unprocessed segments (0.063 seconds)
2020-08-11T23:00:13-07:00 [Error] Unprocessed segment: {
  ErrorCode: "InvalidId",
  Id: "0",
  Message: "Invalid segment. ErrorCode: InvalidId"
}
```

Reply:
Is `debug` level logging enabled in this test?

Reply:
oh, will fix that in https://github.com/aws/aws-xray-daemon/issues/62

Reply:
This PR I think is just a small one to improve the log message (not always say Successfully), it's not a fix for the n^2 logging @willarmiros

Reply:
Ah ok, was confused by the closure of #63